File: 프로그래머스/2/147354. 테이블 해시 함수/테이블 해시 함수.py
def solution(data, col, row_begin, row_end):
    answer = 0
    data.sort(key=lambda x : (x[col-1], -x[0]))
    S = []
    for i in range(row_begin - 1, row_end):
        result = 0
        for j in range(len(data[i])):
            result += data[i][j] % (i+1)
        S.append(result)
    answer = S[0]
    
    for j in range(1, len(S)):
        answer = answer ^ S[j]
    return answer

File: 프로그래머스/2/147354. 테이블 해시 함수/테이블 해시 함수.java
import java.util.*;

class Solution {
    public int solution(int[][] data, int col, int row_begin, int row_end) {
        int answer = 0;
    
        Arrays.sort(data, (o1, o2) -> {
            if (o1[col-1] != o2[col-1]) {
                return Integer.compare(o1[col-1], o2[col-1]);
            }
            return Integer.compare(o2[0], o1[0]);
        });
        
        for (int i = row_begin-1; i < row_end; i++) {
            int total = 0;
            for (int j = 0; j < data[i].length; j++) {
                total += data[i][j] % (i+1);
            }
            if (i == row_begin-1) {
                answer = total;
                continue;
            }
            answer ^= total;
        }
        
        
        return answer;
    }
}